Snowflake.Data.Tests/UnitTests/SFBindUploaderTest.cs

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System;
+using NUnit.Framework;
+using Snowflake.Data.Core;
+
+namespace Snowflake.Data.Tests.UnitTests
+{
+    [TestFixture]
+    class SFBindUploaderTest
+    {
+        private readonly SFBindUploader _bindUploader = new SFBindUploader(null, "test");
+
+        [TestCase(SFDataType.DATE, "0", "1/1/1970")]
+        [TestCase(SFDataType.DATE, "73814400000", "5/4/1972")]
+        [TestCase(SFDataType.DATE, "1709193600000", "2/29/2024")]
+        public void TestCsvDataConversionForDate(SFDataType dbType, string input, string expected)
+        {
+            DateTime dateExpected = DateTime.Parse(expected);
+            DateTime dateActual = DateTime.Parse(_bindUploader.GetCSVData(dbType.ToString(), input));
+            Assert.AreEqual(dateExpected, dateActual);
+        }
+
+        [TestCase(SFDataType.TIME, "0", "00:00:00.000000")]
+        [TestCase(SFDataType.TIME, "100000000", "00:00:00.100000")]
+        [TestCase(SFDataType.TIME, "1000000000", "00:00:01.000000")]
+        [TestCase(SFDataType.TIME, "60123456000", "00:01:00.123456")]
+        [TestCase(SFDataType.TIME, "46801000000000", "13:00:01.000000")]
+        public void TestCsvDataConversionForTime(SFDataType dbType, string input, string expected)
+        {
+            DateTime timeExpected = DateTime.Parse(expected);
+            DateTime timeActual = DateTime.Parse(_bindUploader.GetCSVData(dbType.ToString(), input));
+            Assert.AreEqual(timeExpected, timeActual);
+        }
+        
+        [TestCase(SFDataType.TIMESTAMP_LTZ, "0", "2012-07-01T12:00:00.0000000+02:00")]
+        [TestCase(SFDataType.TIMESTAMP_LTZ, "1341144000000000000", "2012-07-01T12:00:00.0000000+02:00")]
+        [TestCase(SFDataType.TIMESTAMP_LTZ, "352245599987654000", "1981-02-28T23:59:59.9876540+02:00")]
```

Comment:
It will work in CET timezone only? What if someone will run the tests in another timezone?

Reply:
Fixed, upload happens after converting to local tz, during check expected value is also in local tz.

Reply:
Checked manually in 3 different timezones (we can work it out later to swap tz during the test) - US/NY, JPN/Tokyo, PL/WAW